bolts/data/datasets/audio/ecoacoustics_ds.py

```diff
@@ -0,0 +1,121 @@
+"""Ecoacostics dataset provided A. Eldridge et al.
+    Alice Eldridge, Paola Moscoso, Patrice Guyot, & Mika Peck. (2018).
+    Data for "Sounding out Ecoacoustic Metrics: Avian species richness
+    is predicted by acoustic indices in temperate but not tropical
+    habitats" (Final) [Data set].
+    Zenodo. https://doi.org/10.5281/zenodo.1255218
+"""
+from __future__ import annotations
+import os
+from os import listdir
+from os.path import isfile, join
+from pathlib import Path
+from typing import ClassVar, Optional, Union
+import zipfile
+
+import numpy as np
+import pandas as pd
+import torch
+from typing_extensions import Literal
+
+from bolts.data.datasets.audio.base import PBAudioDataset
+from bolts.data.datasets.utils import AudioTform, FileInfo
+
+__all__ = ["EcoacousticsDS"]
+SoundscapeAttr = Literal["habitat", "site"]
+
+
+class EcoacousticsDS(PBAudioDataset):
+    """Dataset for audio data collected in various geographic locations."""
+
+    LABELS_DIR: ClassVar[str] = "AvianID_AcousticIndices"
+    METADATA_FILENAME: ClassVar[str] = "metadata.csv"
+
+    _FILE_INFO: ClassVar[FileInfo] = FileInfo(name="EcoacousticsDS.zip", id="PLACEHOLDER")
+    _BASE_FOLDER: ClassVar[str] = "EcoacousticsDS"
+    _EC_LABELS_FILENAME: ClassVar[str] = "EC_AI.csv"
+    _UK_LABELS_FILENAME: ClassVar[str] = "UK_AI.csv"
+
+    def __init__(
+        self,
+        root: Union[str, Path],
+        *,
+        download: bool = True,
+        target_attr: SoundscapeAttr = "habitat",
+        transform: Optional[AudioTform] = None,
+    ) -> None:
+
+        self.root = Path(root)
+        self.download = download
+        self._base_dir = self.root / self._BASE_FOLDER
+        self.labels_dir = self._base_dir / self.LABELS_DIR
+        self._metadata_path = self._base_dir / "metadata.csv"
+        self.ec_labels_path = self.labels_dir / self._EC_LABELS_FILENAME
+        self.uk_labels_path = self.labels_dir / self._UK_LABELS_FILENAME
+
+        # Check data is downloaded and unzipped.
+        if self.download:
+            pass  # TODO: Implement download util function.
+        elif not self._files_unzipped():
+            raise RuntimeError(
+                f"Data not found at location {self._base_dir.resolve()}. " "Have you downloaded it?"
+            )
+
+        if not self._metadata_path.exists():
+            self._extract_metadata(target_attr)
+
+        self.metadata = pd.read_csv(self._base_dir / "metadata.csv")
+
+        x = self.metadata["filePath"].to_numpy()
+        y = torch.as_tensor(self.metadata[f'{target_attr}_le'])
+        s = None
+
+        super().__init__(x=x, y=y, s=s, transform=transform, audio_dir=self._base_dir)
+
+    def _files_unzipped(self) -> bool:
+        dir = self.root / self._BASE_FOLDER
```

Comment:
```suggestion
        dir_ = self.root / self._BASE_FOLDER
```
`dir` mirrors a python builtin

Reply:
The more you know. Thanks.

Reply:
A better IDE would've probably told me this as well...